src/consensus/replay_tower.zig

```diff
@@ -0,0 +1,2735 @@
+        0,
+        &slot_history,
+    );
+
+    try std.testing.expect(true);
+}
+
+test "adjust lockouts after replay future tower" {
+    var replay_tower = try createTestReplayTower(std.testing.allocator, 10, 0.9);
+    defer replay_tower.deinit(std.testing.allocator);
+
+    try replay_tower.tower.vote_state.votes.append(
+        Lockout{ .slot = 13, .confirmation_count = 1 },
+    );
+
+    try replay_tower.tower.vote_state.votes.append(
+        Lockout{ .slot = 14, .confirmation_count = 1 },
+    );
+
+    const slots = [_]Slot{14};
+    const vote = Vote{
+        .slots = &slots,
+        .hash = Hash.ZEROES,
+        .timestamp = null,
+    };
+
+    replay_tower.last_vote = VoteTransaction{ .vote = vote };
+    replay_tower.tower.initializeRoot(12);
+
+    var slot_history = try createTestSlotHistory(std.testing.allocator);
+    defer slot_history.bits.deinit(std.testing.allocator);
+    slot_history.add(@as(Slot, 0));
+    slot_history.add(@as(Slot, 2));
+
+    var expected_votes = [_]Slot{ 13, 14 };
+
+    try replay_tower.adjustLockoutsAfterReplay(
+        std.testing.allocator,
+        2,
+        &slot_history,
+    );
+
+    const voted_slots = try replay_tower.tower.votedSlots(std.testing.allocator);
+    defer std.testing.allocator.free(voted_slots);
+
+    try std.testing.expectEqual(
+        12,
+        try replay_tower.tower.getRoot(),
+    );
+    try std.testing.expectEqualSlices(
+        Slot,
+        &expected_votes,
+        voted_slots,
+    );
+    try std.testing.expectEqual(
+        14,
+        replay_tower.stray_restored_slot,
+    );
+}
+
+test "is slot confirmed not enough stake failure" {
+    var tower = try createTestReplayTower(std.testing.allocator, 1, 0.67);
+    defer tower.deinit(std.testing.allocator);
+
+    var stakes = AutoHashMapUnmanaged(u64, u64){};
+    defer stakes.deinit(std.testing.allocator);
+    try stakes.ensureTotalCapacity(std.testing.allocator, 1);
+
+    stakes.putAssumeCapacity(0, 1);
+
+    const result = isSlotConfirmed(&tower, 0, &stakes, 2);
+    try std.testing.expect(!result);
+}
+
+test "is slot confirmed unknown slot" {
+    var tower = try createTestReplayTower(std.testing.allocator, 1, 0.67);
+    defer tower.deinit(std.testing.allocator);
+
+    var stakes = AutoHashMapUnmanaged(u64, u64){};
+    defer stakes.deinit(std.testing.allocator);
+
+    const result = isSlotConfirmed(&tower, 0, &stakes, 2);
+    try std.testing.expect(!result);
+}
+
+test "is slot confirmed pass" {
+    var tower = try createTestReplayTower(std.testing.allocator, 1, 0.67);
+    defer tower.deinit(std.testing.allocator);
+
+    var stakes = AutoHashMapUnmanaged(u64, u64){};
+    defer stakes.deinit(std.testing.allocator);
+    try stakes.ensureTotalCapacity(std.testing.allocator, 1);
+
+    stakes.putAssumeCapacity(0, 2);
+
+    const result = isSlotConfirmed(&tower, 0, &stakes, 2);
+    try std.testing.expect(result);
+}
+
+test "default tower has no stray last vote" {
```

Comment:
Sorry I missed this in the PR. Only noticing now that the `test:` prefix were removed. 

Is there any benefit to not having the prefix?

One benefit to having it, is that it comes in handy when filtering tests to run. ie: `zig build test -Dfilter="tower:"`.

Also, although not exactly matching what is in the style guide, https://github.com/Syndica/sig/blob/f66a2c7a81b8052d4645c372371686a911b6e6d1/docs/docusaurus/docs/contributing/style-guide.mdx?plain=1#L142 but it does come close.